File: backend/app/schemas/default_users.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class DefaultUserBase(BaseModel):
    user_id: int
    position: int = 0


class DefaultUserCreate(DefaultUserBase):
    pass


class DefaultUserUpdate(BaseModel):
    position: Optional[int] = None


class DefaultUser(DefaultUserBase):
    id: int
    created_at: datetime
    
    class Config:
        from_attributes = True


class DefaultUserWithUser(DefaultUser):
    user_name: Optional[str] = None
    user_email: Optional[str] = None


class DefaultUsersReorder(BaseModel):
    """Схема для изменения порядка дефолтных пользователей"""
    user_ids: list[int]  # Список ID пользователей в новом порядке

File: frontend/src/types/defaultUsers.ts
export interface DefaultUser {
  id: number;
  user_id: number;
  position: number;
  created_at: string;
}

export interface DefaultUserWithUser extends DefaultUser {
  user_name: string | null;
  user_email: string | null;
}

export interface DefaultUserCreate {
  user_id: number;
  position?: number;
}

export interface DefaultUsersReorder {
  user_ids: number[];
}
